test/test_utils.py

```diff
@@ -75,23 +75,24 @@ def run_cmd(command, cwd, verbose = False, expect_success = True):
         return result.stderr
 
 def execute_command(host_bin, control_protocol, cwd, cmd_name, cmd_map_path = None, cmd_vals = None, expect_success = True):
-    
     command = str(host_bin) + " -u " + control_protocol + " " + cmd_name
     if cmd_map_path:
         print(f"cmd_map_path in execute_command() is {cmd_map_path}")
         command = str(host_bin) + " -u " + control_protocol + " -cmp " + cmd_map_path + " " + cmd_name
     if cmd_vals != None:
         cmd_write = command + " " + ' '.join(str(val) for val in cmd_vals)
         run_cmd(cmd_write, cwd, True, expect_success)
+        expect_success = True # Set this to true so that the next run_cmd will not fail
```

Comment:
I'm not sure why this line has been added. In case of a read command, it will override expect_success variable

Reply:
This is needed because the call to `run_cmd()` just below always expects "expect_success = True" if `cmd_vals` is set. I found it quite an unusual implementation choice, and I didn't want to change too much of the original code, so I chose the simplest fix.